osdconfig/api.go

```diff
@@ -185,39 +189,74 @@ func (manager *configManager) SetNodeConf(config *NodeConfig) error {
 	defer manager.Unlock()
 
 	if config == nil {
-		return InputErr
+		return ErrorInput
 	}
 
 	if len(config.NodeId) == 0 {
-		return InputErr
+		return ErrorInput
 	}
 
-	// push into kvdb
+	// push node data into kvdb
 	if _, err := manager.cc.Put(getNodeKeyFromNodeID(config.NodeId), config, 0); err != nil {
 		return err
 	}
+
+	// update node index
+	list, err := manager.GetNodeList()
+	if err != nil {
+		list = make([]string, 0, 0)
```

Comment:
This is not necessary. A `nil` list is acceptable for append. See: https://play.golang.org/p/7kh_x-u-9jr

Reply:
will do